File: app/src/main/java/com/mdgd/pokemon/bg/UploadWorker.java
package com.mdgd.pokemon.bg;


import android.content.Context;

import androidx.annotation.NonNull;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

public class UploadWorker extends Worker {

    private final LoadPokemonsContract.ServiceModel model;

    public UploadWorker(@NonNull Context context, @NonNull WorkerParameters params) {
        super(context, params);
        model = new PokemonsLoadingModelFactory().create();
    }

    @Override
    public Result doWork() {
        model.load();
        return Result.success();
    }
}

File: app/src/main/java/com/mdgd/pokemon/bg/UploadWorker.kt
package com.mdgd.pokemon.bg

import android.content.Context
import androidx.hilt.work.HiltWorker
import androidx.work.Worker
import androidx.work.WorkerParameters
import com.mdgd.pokemon.models.cache.Cache
import com.mdgd.pokemon.models.repo.PokemonsRepo
import dagger.assisted.Assisted
import dagger.assisted.AssistedInject

@HiltWorker
class UploadWorker @AssistedInject constructor(
    @Assisted context: Context, @Assisted params: WorkerParameters,
    repo: PokemonsRepo, cache: Cache
) : Worker(context, params) {
    private val model: ServiceModel = LoadPokemonsModel(repo, cache)

    override fun doWork(): Result {
        model.load()
        return Result.success()
    }
}
